lib/plugins/wallet/cardano/cardano.js

```diff
@@ -0,0 +1,85 @@
+const { walletId } = require('./mnemonic')
+const cardanoApi = require('./client')
+
+// TODO: work with BN here instead of api.js?
+
+async function checkCryptoCode(cryptoCode) {
+  if (cryptoCode !== 'ADA') {
+    return Promise.reject(new Error(`Unsupported crypto: ${cryptoCode}`))
+  }
+
+  return Promise.resolve()
+}
+
+/**
+ * TODO: instead of using walletId function we can
+ *       inject walletId/index into account object
+ *       in wallet.js - it's better?
+ */
+async function balance(account, cryptoCode) {
+  return checkCryptoCode(cryptoCode)
+    .then(() => walletId())
```

Comment:
you could do `.then(walletId)`

Reply:
i agree with ppershing about visible arguments